lib/matplotlib/axes/_axes.py

```diff
@@ -3885,7 +3885,8 @@ def boxplot(self, x, notch=None, sym=None, vert=None, whis=None,
             boxes are drawn with Patch artists.
 
         labels : sequence, optional
-            Labels for each dataset (one per dataset).
+            Labels for each dataset (one per dataset). These are used for
+            x-ticks; *not* for legend entries.
```

Comment:
```suggestion
            labels on the x-axis; *not* for legend entries.
```
or `x-tick labels` would work too.